presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -83,6 +85,58 @@ private PlanNodeStatsEstimate filterForUnknownExpression()
         }
 
         @Override
+        protected PlanNodeStatsEstimate visitLogicalBinaryExpression(LogicalBinaryExpression node, Void context)
+        {
+            PlanNodeStatsEstimate left = process(node.getLeft());
+            PlanNodeStatsEstimate right = process(node.getRight());
+            switch (node.getType()) {
+                case AND:
+                    return intersectStats(left, right);
+                break;
+                case OR:
+                    return unionStats(left, right);
+                break;
+            }
+        }
+
+        private PlanNodeStatsEstimate unionStats(PlanNodeStatsEstimate left, PlanNodeStatsEstimate right)
+        {
+            return input; // FIXME - implement this
```

Comment:
It does not look like reasonable default.
As this commit does not implement union drop this method.
Drop `AND` branch in case in `visitLogicalBinaryExpression`
And add `default: filterStatsByFactor(0.5)` or `default: super.visitLogicalBinaryExpression(node, context)`